### PR TITLE
Incs 5657 create generic react component for app configuration for starting screen

### PR DESCRIPTION
**Pull request by lcaglar**
Adding a generic landing page react component with reference to these design docs: https://www.figma.com/file/9PLctPRw78hehDvxB86vqC/Better-Configs?type=design&node-id=1-2&mode=design&t=B2KweNF920llL73a-0

Page follows the GH4J structure: 
- Explain what needs to be configured
- Explain what permissions are needed
- Links to guides/helpers

Screenshot 2024-01-24 at 4 25 39 pm

Screenshot 2024-01-19 at 9 11 10 am

### CODE REVIEW

File: spa/package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "@atlaskit/badge": "^15.2.1",
+    "@atlaskit/dropdown-menu": "^12.1.10",
+    "@atlaskit/icon": "^22.0.0",
+    "@atlaskit/lozenge": "^11.5.0",
+    "@atlaskit/tokens": "^1.33.0",
+    "@sentry/react": "^7.93.0",
+    "@testing-library/jest-dom": "^5.17.0",
+    "@testing-library/react": "^13.4.0",
+    "@testing-library/user-event": "^13.5.0",
+    "axios": "^1.6.5",
```
Comment (ttayeb-atlassian) [inline spa/package.json:17]: I dont think we're using axios in the app, can you please check if these packages are used in the spa package and remove them from dependancies if they arent 🙏

File: spa/public/logo192.png
Comment (ttayeb-atlassian) [inline spa/public/logo192.png:1]: can we remove these logos that are unused 🙏

File: spa/public/favicon.ico
Comment (ttayeb-atlassian) [inline spa/public/favicon.ico:1]: do we need to set this favicon to something? i think we'll need to set it to whatever the github for jira apps favicon is - can you check with sarah?

File: spa/src/App.test.tsx
Comment (ttayeb-atlassian) [inline spa/src/App.test.tsx:1]: we should update this test to test your changes or remove it for now if we are adding tests later

File: spa/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+root.render(
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
```
Comment (krazziekay) [inline spa/src/index.tsx:19]: Please remove this too, since we're not using this.

File: spa/src/reportWebVitals.ts
Comment (krazziekay) [inline spa/src/reportWebVitals.ts:1]: Don't think we'll be using this too, please remove this as well.

File: spa/src/index.css
Comment (ttayeb-atlassian) [inline spa/src/index.css:1]: we should remove this

File: yarn.lock [hunk 3]
```diff
@@ -2,40 +2,462 @@
 # yarn lockfile v1
 
 
+"@atlaskit/analytics-next-stable-react-context@1.0.1":
+  version "1.0.1"
+  resolved "https://packages.atlassian.com/api/npm/npm-remote/@atlaskit/analytics-next-stable-react-context/-/analytics-next-stable-react-context-1.0.1.tgz#93c4c7a1a35f4ab606d727af443f49f0842fd96f"
```
Comment (krazziekay) [inline yarn.lock:7]: `packages.atlassian.com` can't be accessed by non-Atlassian devs. Will need to change the registry to `registry.npmjs.org`.

File: spa/src/rest-interfaces/index.ts
Comment (krazziekay) [inline spa/src/rest-interfaces/index.ts:1]: These types are all specific to Github and I don't think we need these types for this case.

**Review by krazziekay — CHANGES_REQUESTED** [review]

File: spa/src/pages/start-connection/ExampleComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	padding: ${token("space.100")} ${token("space.150")};
+	position: absolute;
+	top: -22px;
+`;
+
+// The ExampleComponent component displays a inline tooltip modal with custom text
+// Content of the inline dialog can be defined here
+export const ExampleComponent = () => {
+	const InlineDialogContent = () => (
+		<>
```
Comment (ttayeb-atlassian) [inline spa/src/pages/start-connection/ExampleComponent.tsx:35]: i think the this file needs to be formatting, the indentation is off

File: spa/src/rest-interfaces/index.ts
Comment (ttayeb-atlassian) [inline spa/src/rest-interfaces/index.ts:1]: we should remove this if unused

File: src/spa-proxy.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { Express } from "express";
+import httpProxy from "http-proxy";
+//import { isNodeDev } from "./utils/is-node-env";
```
Comment (ttayeb-atlassian) [inline src/spa-proxy.ts:3]: we should remove this comment

File: spa/src/common/Wrapper.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+const wrapperCenterStyle = css`
+	margin: 0 auto;
+	max-width: 580px;
+	height: calc(100vh - ${navHeight * 2}px);
+	display: flex;
+	flex-direction: column;
+	justify-content: center;
+`;
+
+export const Wrapper = (attr: {
```
Comment (ttayeb-atlassian) [inline spa/src/common/Wrapper.tsx:20]: this looks like a wrapper for the page components? if it is can we rename this to PageWrapper

File: spa/src/common/Wrapper.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+	children?: ReactNode | undefined;
+}) => {
+	return (
+		<div css={wrapperStyle}>
+			{!attr.hideClosedBtn && (
+				<Button
+					style={{ float: "right" }}
+					iconBefore={<CrossIcon label="Close" size="medium" />}
+					appearance="subtle"
+					//onClick={navigateToHomePage}
```
Comment (ttayeb-atlassian) [inline spa/src/common/Wrapper.tsx:31]: should this be commented out?

File: spa/package.json (hunk 1 above)
Comment (lcaglar) [reply spa/package.json:17]: Axios isn't used, I've removed it!

File: spa/public/favicon.ico
Comment (lcaglar) [reply spa/public/favicon.ico:1]: The GH4J app doesn't seem to set the favicon to anything but yeah I'll ask Sarah to be sure

File: spa/public/logo192.png
Comment (lcaglar) [reply spa/public/logo192.png:1]: Done!

File: spa/src/App.test.tsx
Comment (lcaglar) [reply spa/src/App.test.tsx:1]: I've removed this file for now

File: spa/README.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+For local instances, a proxy server has been created to run this locally. Three procedures must be run in order to view the frontend on Jira:
+
+1.run the node app:
+atlassian-connect-example-app-node % yarn start
+
+2.run the react app:
+spa % yarn start
+
+3.run ngrock:
```
Comment (krazziekay) [inline spa/README.md:15]: Typo `ngrok`
And please do not share your own domain in such public doc.

File: spa/README.md
Comment (krazziekay) [inline spa/README.md:1]: While this makes sense to us, I'd still suggest to run this text by your team's content designer, cause this is where every 3P dev is gonna land at first.

File: spa/package.json [hunk 9]
```diff
@@ -0,0 +1,66 @@
+{
+  "name": "my-app",
```
Comment (krazziekay) [inline spa/package.json:2]: \nit Should come up with a better name than `my-app`.

File: spa/src/components/ExampleComponent/index.tsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+	padding: ${token("space.100")} ${token("space.150")};
+	position: absolute;
+	top: -22px;
+`;
+
+// The ExampleComponent component displays an inline tooltip modal with custom text.
+// The content of the inline dialog can be defined here.
+export const ExampleComponent = () => {
+	const InlineDialogContent = () => (
+		<>
```
Comment (krazziekay) [inline spa/src/components/ExampleComponent/index.tsx:35]: \nit Don't need these React Fragments.

**Review by krazziekay — COMMENTED** [review]
Minor comments

File: spa/README.md
Comment (lcaglar) [reply spa/README.md:1]: I've added some more info to it while I wait to hear back from the designer!

**Review by krazziekay — APPROVED** [review]
Good work 👍

File: spa/README.md
Comment (krazziekay) [reply spa/README.md:1]: If it takes too long, then please do that in a separate PR. The rest of the stuff looks good.

Comment (lcaglar) [reply spa/README.md:1]: Haven't heard back so I'll just do that, thanks

**Review by krazziekay — APPROVED** [review]

**Pull request merged**